org.mixedrealitytoolkit.uxcore/Button/PressableButton.cs

```diff
@@ -142,12 +147,30 @@ public SpaceMode DistanceSpaceMode
         public bool RejectZRollOff { get => rejectZRollOff; set => rejectZRollOff = value; }
 
         /// <summary>
-        ///  Speed for extending the moving button visuals when selected by a non-touch source.
+        /// Speed for extending the moving button visuals when selected by a non-touch source.
         /// </summary>
         [SerializeField]
         [Tooltip("Speed for extending the moving button visuals when selected by a non-touch source.")]
         private float extendSpeed = 0.5f;
 
+        /// <summary>
+        /// Array with Components (set in Editor) that will be enabled/disabled based on proximity and hovering.
+        /// </summary>
+        [field: SerializeField, Tooltip("Array with Components (set in Editor) that will be enabled/disabled based on proximity and hovering.")]
+        public Component[] ProximityHoveringDynamicComponents { get; set; } = null;
+
+        /// <summary>
+        /// Fired when an interactable starts hovering or becomes in proximity of the button.
+        /// </summary>
+        [field: SerializeField, Tooltip("Fired when an interactable starts hovering or becomes in proximity of the button.")]
```

Comment:
We should avoid having multiple arguments in an event.    Please take a look at the hover events for the pattern we should use.   Look at the firstHoverEntered and lastHoverExited events (https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.0/api/UnityEngine.XR.Interaction.Toolkit.XRBaseInteractable.html).   Your event should have a single argument. For example, the HoverEnterEvent as a single HoverEnterEventArgs (https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.0/api/UnityEngine.XR.Interaction.Toolkit.HoverEnterEvent.html).

Also, for this scenario, we're more interested in the first and last proximity hovers ...  since we are only wanting to hide the front plates.  
 
So here's what I suggest.  Add two event types and args types

```cs
/// <summary>
/// The event that is called only when the first Interactor begins a proximity hover
/// over this Interactable. Subsequent Interactors that begin a proximity over this Interactable
/// will not cause this event to be invoked as long as any others are still proximity hovering.
/// </summary>
[Serializable]
public sealed class ProximityHoverEnterEvent : UnityEvent<ProximityHoverEnterEventArgs>

/// <summary>
/// Event data associated with the event when an Interactor initiates a proximity hover over an Interactable.
/// </summary>
public class ProximityHoverEnterEventArgs 
{
    /// <summary>
    /// The Interactable associated with the interaction event.
    /// </summary>
    public new INearInteractionModeTarget interactableObject
    {
        get => (IXRHoverInteractable)base.interactableObject;
        set => base.interactableObject = value;
    }
 }

/// <summary>
/// The event that is called only when the last remaining proximity hovering Interactor
/// ends proximity hovering over this Interactable.
/// </summary>
[Serializable]
public sealed class ProximityHoverExitEvent : UnityEvent<ProximityHoverExitEventArgs>

/// <summary>
/// Event data associated with the event when an Interactor initiates a proximity hover over an Interactable.
/// </summary>
public class ProximityHoverExitEventArgs 
{
    /// <summary>
    /// The Interactable associated with the interaction event.
    /// </summary>
    public new INearInteractionModeTarget interactableObject
    {
        get => (IXRHoverInteractable)base.interactableObject;
        set => base.interactableObject = value;
    }
 }
```       
Then add two events like this

```cs
/// <summary>
/// The event that is called only when the first Interactor begins a proximity hover
/// over this Interactable. Subsequent Interactors that begin a proximity over this Interactable
/// will not cause this event to be invoked as long as any others are still proximity hovering.
/// </summary>
[field: SerializeField, Tooltip("The event that is called only when the first Interactor begins a proximity hover over this Interactable..")]
public ProximityHoverEnterEvent FirstProximityHoverEntered { get; set } = new ProximityHoverEnterEvent();

/// <summary>
/// The event that is called only when the last remaining proximity hovering Interactor
/// ends proximity hovering over this Interactable.
/// </summary>
[field: SerializeField, Tooltip("The event that is called only when the last remaining proximity hovering Interactor ends proximity hovering over this Interactable.")]
public ProximityHoverEnterEvent LastProximityHoverEntered { get; set } = new ProximityHoverExitEvent();
```

Then property for querying proximity hover state.  Like this?

```cs
public bool IsProximityHovered { get; private set; } = false;
```

Then add a function to fire events as needed. Maybe like this?

```cs
private void UpdateProximityHovered()
{
   bool oldValue = IsProximityHovered;
   IsProximityHovered = IsHovered || activeCollidersWithInteractor.Count > 0;

   if (oldValue != IsProximityHovered)
   {
        if (IsProximityHovered) 
        {
              // Fire FirstProximityHoverEntered 
        }
        else
        {
            // Fire LastProximityHoverEntered
        }       
   }
}
```

Reply:
Thank you so much for the detailed feedback.  I have attended your suggestions.  I adjusted the names a little bit to reflect that the events respond to a combined Proximity + Hovering state.